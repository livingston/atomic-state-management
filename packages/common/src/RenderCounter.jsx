import { useRef, useEffect } from 'react';

function RenderCounter() {
  const ref = useRef(null);
  useEffect(() => {
    if (ref.current != null) {
      ref.current.textContent = Number(ref.current.textContent || '0') + 1;
    }
  });

  return (
    <span
      className="render-counter absolute h-6 w-6 inline-flex items-center justify-center bg-red-600 bg-opacity-70 cursor-default text-center text-white"
      ref={ref}
    />
  );
}

export default RenderCounter;
