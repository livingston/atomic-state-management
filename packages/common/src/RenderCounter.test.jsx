import { render } from '@testing-library/react';
import RenderCounter from './RenderCounter';

describe('RenderCounter', () => {
  it('should render the component', () => {
    const { container } = render(<RenderCounter />);

    expect(container.textContent).toBe('1');
    expect(container).toMatchSnapshot();
  });

  it('should increment the count on each render', () => {
    const { container, rerender } = render(<RenderCounter />);

    expect(container.textContent).toBe('1');

    rerender(<RenderCounter />);
    expect(container.textContent).toBe('2');

    rerender(<RenderCounter />);
    expect(container.textContent).toBe('3');
  });
});
