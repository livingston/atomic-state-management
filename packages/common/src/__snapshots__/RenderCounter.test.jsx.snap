// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderCounter should render the component 1`] = `
<div>
  <span
    class="render-counter absolute h-6 w-6 inline-flex items-center justify-center bg-red-600 bg-opacity-70 cursor-default text-center text-white"
  >
    1
  </span>
</div>
`;
